test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -1019,8 +1019,12 @@ def run_cmd(self, hosts=None, cmd=None, sudo=False, stdin=None,
 
             if as_script:
                 # must pass as_script=False otherwise it will loop infinite
-                self.rm(hostname, path=_script, as_script=False,
-                        level=level, runas=runas)
+                if platform == 'shasta' and not runas:
+                    self.rm(hostname, path=_script, as_script=False,
+                            level=level, runas=runas)
+                else:
+                    self.rm(hostname, path=_script, as_script=False,
+                            level=level)
```

Comment:
OK, this looks wrong.  First you took runas=runas out of the normal case.  Second, you said 'not runas' and then added runas=runas.  Won't it be empty?

Reply:
Interesting, this is the commit i force-pushed but the changes still didn't get added. I must not have committed them.
You're right, it should be `if platform == 'shasta' and runas:`